src/tests/ftest/container/query_attribute.py

```diff
@@ -0,0 +1,205 @@
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a ContainerQueryAttribute object."""
+        super(ContainerQueryAttributeTest, self).__init__(*args, **kwargs)
+        self.expected_pool_uuid = None
+        self.expected_cont_uuid = None
+        self.sr = None
+        self.daos_cmd = None
+    
+    def create_pool_container(self):
+        """Create a pool and a container in the pool.
+
+        Save some variables so that we can use them in the tests.
+        """
+        pool_create_result = self.get_dmg_command().get_output(
+            "pool_create", scm_size="1G")
+        self.expected_pool_uuid = pool_create_result[0]
+        self.sr = pool_create_result[1]
+        self.daos_cmd = DaosCommand(self.bin)
+        self.expected_cont_uuid = self.daos_cmd.get_output(
+            "container_create", pool=self.expected_pool_uuid, svc=self.sr)[0]
+
+    def test_container_query_attr(self):
+        """JIRA ID: DAOS-4640
+
+        Test Description:
+            Test daos container query and attribute commands as described
+            above.
+
+        Use Cases:
+            Test container query, set-attr, get-attr, and list-attrs.
+
+        :avocado: tags=all,pool,small,full_regression,cont_query_attr
+        """
+        # Test pool query.
+        self.create_pool_container()
+        # Call daos container query, obtain pool and container UUID, and
+        # compare against those used when creating the pool and the container.
+        kwargs = {
+            "pool": self.expected_pool_uuid,
+            "svc": self.sr,
+            "cont": self.expected_cont_uuid
+        }
+        query_output = self.daos_cmd.get_output("container_query", **kwargs)[0]
+        actual_pool_uuid = query_output[0]
+        actual_cont_uuid = query_output[1]
+        self.assertEqual(actual_pool_uuid, self.expected_pool_uuid)
+        self.assertEqual(actual_cont_uuid, self.expected_cont_uuid)
+
+        # Test container set-attr, get-attr, and list-attrs with different
+        # types of characters.
+        test_strings = [
+            "abcd",
+            "1234",
+            "abc123",
+            "abcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghij",
+            # Characters that don't require backslash. The backslashes in here
+            # are required for the code to work, but not by daos.
+            "~@#$%^*-=_+[]\{\}:/?,.",
+            # Characters that require backslash.
+            "\`\&\(\)\\\;\\'\\\"\!\<\>",
+            # Characters that include space.
+            "\"aa bb\""]
+        # We added backslashes for the code to work, but get-attr output
+        # doesn't contain them, so prepare the expected output that doesn't
+        # include backslashes.
+        escape_to_not = {}
+        escape_to_not[test_strings[-3]] = "~@#$%^*-=_+[]{}:/?,."
+        # We still need a backslash before the double quote for the code to
+        # work.
+        escape_to_not[test_strings[-2]] = "`&()\;'\"!<>"
```

Comment:
(lint) '\;'. String constant might be missing an r prefix.

Reply:
I tested with r prefix, but it didn't work.